### DISCUSSION TITLE
Is there a way to define curl timeouts?

### DISCUSSION THREAD

**Discussion by studersi** [OpenIDC/mod_auth_openidc#1189] [Q&A] [answered]
There are timeouts in place to limit the duration of back-channel request. In case of DNS lookup failure (and possibly other issues with the HTTP request), the timeout is set to 10s by default:
```
oidc_util_http_call: curl_easy_perform failed with a timeout for https://idp/token/endpoint: [Resolving timed out after 10000 milliseconds]; won't retry
```

Is there a way to set this timeout to a lower value?

**Comment by zandbelt** [comment, marked as answer]
yes, since 2.4.15 there is:
https://github.com/OpenIDC/mod_auth_openidc/blob/v2.4.15.3/auth_openidc.conf#L865C1-L871C85

**Reply by studersi** [reply]
Perfect, thank you!
